ldms/src/sampler/file_exporter/ldms-sampler_file_exporter.rst

```diff
@@ -0,0 +1,239 @@
+
+   **plugin**\ *=file_exporter*
+      This must be *file_exporter*
+
+   **dir_path**\ *=<PATH>*
+      The path to the directory from which all files will be imported.
+
+   **uid**\ *=<user-id>*
+      The user-id that will be placed in the meta-data of the metric set. This is only
+      honored if the **ldmsd** is owned by root, otherwise it is the value returned
+      by *geteuid()*
+
+   **gid**\ *=<group-id>*
+      The group-id that will be placed in the meta-data of the metric set. This is only
+      honored if the **ldmsd** is owned by root, otherwise it is the value returned
+      by *getegid()*
+
+   **instance**\ *=<format-specifier>*
+      The format specifier is a string that specifies how sets are named.
+      There are two vectors in which the *instance name* should be unique.
+      First, the *instance name* needs to be unique among other sets
+      produced in the monitored system. This is to ensure that
+      there are no name collisions among  other set names from different
+      producers
+
+      The format specifiers are as follows:
+
+      - **%H** The value of the *HOSTNAME* environment variable
+      - **%P** The value of the *producer* configuration attribute
+      - **%p** The plugin name, i.e., 'file_exporter'
+      - **%C** The configuration name, i.e., the value specified by
+        the *name* attribute in the **load** commmand
+      - **%S** The value of the *schema* configuration attribute. The
+        default value for the schema is: <plugin-name>'.'<config-name>'.'<basename(file-path)>'
+      - **%U** The set owner user-id, or the value of the *uid* configuration parameter if the **ldmsd** is a owned by *root*
+      - **%G** The set owner group-id, or the value of the *gid* configuration parameter if the **ldmsd** is a owned by *root*
+      - **%F** The base name of the file from which data is being imported
+
+   For example:
+
+   .. code-block:: text
+
+      config name=fe2 instance="%H/%C/%U/%G/%p.%F"
+
+   Produces something like: "calaluna/fe2/1000/1000/file_exporter.smart2"
+
+
+   - **schema**\ *=<format-specifier>*
+      A format specifier that controlls the naming of each set schema.
+      The schema name *MUST* be unique within the sampler, but *SHOULD* be
+      unique across the cluster for identically formatted sets to allow for
+      reasonable storage and analysis.
+
+    The format specifiers for **schema** are identical to those for **instance**.
+
+   **<SAMPLER_BASE_OPTIONS>**
+      Please see :ref:`ldms_sampler_base(7) <ldms_sampler_base>` for sampler_base options.
+
+NOTES
+=====
+
+Files that contain parsing errors will result in the logging of error
+messages. The file will be skipped and importing will continue with
+the next file.
+
+FILES
+=====
+
+Files imported are self describing, i.e., the rules that govern how
+the data is imported into a metric set is specified in the file
+itself.
+
+Overall the file consists of three main sections: the HELP section,
```

Comment:
What exactly does "section" mean? Do these lines need to appear on particular lines in the file? Is the order significant? Can we have more than one HELP and TYPE line? If so, what are the ordering requirements there?